connectors/sources/google_drive.py

```diff
@@ -142,11 +155,14 @@ async def get_all_folders(self):
 
         return folders
 
-    async def list_files(self, fetch_permissions=False):
+    async def list_files(
```

Comment:
I don't think passing the `syncing_shared_drives` arg is best approach here since we don't always know in advance that we are syncing shared drive.

For example we could share a drive with a service account and perform a standard sync (not with domain delegation), in that case it should see that for a given file we have:
- `trashed=true` and `trashedTime` (and we should actually delete the file if the trashedTime is bigger than last sync timestamp) in that case 
- if `trashedTime` property is missing (meaning the service account was shared with personal drive) then we always delete

So IMO `syncing_shared_drives` is not needed, instead we should add logic to handle deleted based on `trashed` and `trashedTime` properties, or you can even fetch `driveId` from file property (https://developers.google.com/drive/api/reference/rest/v3/files#File) to detect when the file is from a shared drive

Reply:
The issue is that I have service account, that has access only to shared drive. Even though our sync logic should be detecting "incremental deletes" with `trashedTime` property, I see that all trashed items from shared drive are always deleted in the incremental sync -> that should be improved.